boards/shields/huatian_mt9m114/huatian_mt9m114.overlay

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/{
+	chosen {
+		zephyr,camera = &nxp_csi;
+	};
+};
+
+&zephyr_cam_i2c {
+	mt9m114: mt9m114@48 {
+		compatible = "aptina,mt9m114";
+		reg = <0x48>;
+
+		port {
+			mt9m114_ep_out: endpoint {
+				remote-endpoint = <&csi_ep_in>;
+			};
+		};
+	};
+};
+
+&nxp_csi {
+	status = "okay";
+	sdev = <&mt9m114>;
+
+	port {
+		csi_ep_in: endpoint {
+			remote-endpoint = <&mt9m114_ep_out>;
```

Comment:
how is this building, isn't it circular?

Reply:
AFAIK circular phandles are not supported by zephyr DT tooling, so this overlay seems invalid for zephyr

Reply:
@decsny : I think it is misunderstood here. I know that Zephyr tool will check phandle dependencies at compile time and circular phandles are not allowed. But this is **not** the case. These are not phandles but "endpoints" and "remote-endpoints".

It is **not** me who introduced these into Zephyr. In this PR, I just moved existed things from devicetree to an overlay. Please see [this commit](https://github.com/zephyrproject-rtos/zephyr/pull/72435/commits/affe9f18fa46673a941a634f1138f906d74cd4dd) in this same PR.

In fact, "endpoints" and "remote-endpoints" concepts are adopted into Zephyr and it is not limited to video, it is used for display too, please see @MaureenHelm 's commit message [here](https://github.com/zephyrproject-rtos/zephyr/commit/54047dc59a4efbad2e5f11eb12e25dd5009beb3c).
The video APIs by @loicpoulain also uses it.

And you can see as well, it actually builds. This is a good concept that helps to avoid circular phandles or phandle reference init order "chicken-egg" issue as well as other things. It just needs an backend implementation in the future.

Reply:
it's not "checked" by zephyr, it just isn't able to be handled, it would be best if we could implement the support for it in the DT tooling but it is a big project to do so

Reply:
okay, I actually don't get what's going on here, I will see what is going on with these overlays because maybe my understanding is wrong, because I would have thought none of these overlays can build

Reply:
Yes, the overlay actually builds and the [mimxrt1064_evk.dts](https://github.com/zephyrproject-rtos/zephyr/pull/72435/commits/affe9f18fa46673a941a634f1138f906d74cd4dd#diff-4da10656a01fd2ed97c74286e07973da6e6c93574b979e4ecf02d2b0ed100e85) also builds with these nodes defined like that before my commit.

The discussion about endpoint / circular dependency is good but deserve another topic for it I think.

This tiny / trivial PR just moves the camera nodes which are _already defined in the mimxrt1064_evk.dts_ to a separate overlay because, as said in the commit message:

"The mt9m114 camera module is not always attached to this board. Camera stuffs should go into a seperate overlay / shield. Also, the CSI should be enabled only if there is a camera sensor attached to it."

Reply:
@decsny : If you think "port/endpoint/remote" concept should not be in Zephyr for the time being (as it is not a good thing for Zephyr and it is currently not implemented). We (who?) can submit a PR to remove them all from Zephyr (video API need to be changes as well).

BTW, this PR simply moves existing camera node from mimxrt1064_evk.dts to a separate overlay ... Could we move the PR forward ?

Reply:
> AFAIK circular phandles are not supported by zephyr DT tooling, so this overlay seems invalid for zephyr

This was further characterized recently:
https://github.com/zephyrproject-rtos/zephyr/issues/57708

I tried to go to the bottom of the implication of a `remote-endpoint` in Zephyr here:
https://github.com/zephyrproject-rtos/zephyr/issues/72311#issuecomment-2095521208

Reply:
@decsny : Hi, is it ok now with my explanations about the PR or there are somethings you want me to change ?
Thanks

Reply:
> > AFAIK circular phandles are not supported by zephyr DT tooling, so this overlay seems invalid for zephyr
> 
> This was further characterized recently: #57708
> 
> I tried to go to the bottom of the implication of a `remote-endpoint` in Zephyr here: #72311

thanks for this information, it's exactly what I needed I think, I will look into this further soon to see what is going on here